src/pages/gen2/build-a-backend/data/connect-to-existing-database/index.mdx

```diff
@@ -0,0 +1,410 @@
+
+</Callout>
+
+This creates a new **schema.rds.ts** with a schema reflecting the types of your database. Import the schema to your **amplify/data/resource.ts** file and
+
+```ts
+import { a, defineData } from '@aws-amplify/backend';
+// highlight-next-line
+import { schema as auroraSchema } from './schema.rds.ts';
+
+// Relational database sources can coexist with DynamoDB tables managed by Amplify.
+const schema = a.schema({
+  Todo: a.model({
+    content: a.string(),
+    isDone: a.boolean()
+  })
+});
+
+// Use the .combine() operator to stitch together the models backed by DynamoDB
+// and the models backed by Postgres or MySQL databases.
+// highlight-next-line
+const combinedSchema = a.schema.combine([schema, auroraSchema]);
+
+export type Schema = typeof combinedSchema;
+export const data = defineData({
+  // highlight-next-line
+  schema: combinedSchema
+});
+```
+
+## Step 3 - Add authorization rules
+
+```ts
+import { a, defineData } from '@aws-amplify/backend'
+import { schema as rdsSchema } from './schema.rds.ts'
+
+// Add an authorization rule to the schema
+// highlight-next-line
+const modifiedRds = rdsSchema.setAuthorization((models) => [
+  models.todo.authorization([a.allow.public()]),
+  models.todo.fields.content.authorization([a.allow.private()]),
+]);
+
+const schema = a.schema({
+  Post: a.model({
+    title: a.string()
+  }).authorization([a.allow.public()])
+})
+
+const combinedSchema = a.schema.combine([
+  schema,
+  modifiedRds
+])
+
+export type Schema = typeof schema
+export const data = defineData({
+  schema: combinedSchema
+})
+```
+
+## Step 4 - Deploy your Data resources using the cloud sandbox
+
+Finally, you can now validate your Data resources with your cloud sandbox:
+
+```bash
+npx amplify sandbox
+```
+
+## Rename generated models and fields
+
+```ts
+import { a, defineData } from '@aws-amplify/backend'
+import { schema as rdsSchema } from './schema.rds.ts'
+
+// Rename models or fields to be more idiomatic for frontend code
+// highlight-start
+const modifiedRds = rdsSchema.renameModels(() => [
+  ['todo', 'Todo']
+]).renameModelFields((models) => [
+    models.Todo.renameFields(() => [
+      ['title', 'content'],
+  ]),
+])
+
+const schema = a.schema({
+  Post: a.model({
+    title: a.string()
+  }).authorization([a.allow.public()])
+})
+
+const combinedSchema = a.schema.combine([
+  schema,
+  modifiedRds
+])
+
+
+export type Schema = typeof schema
+export const data = defineData({
+  schema: combinedSchema
+})
+```
+
+Step 5 - Configure the `amplify.yml` TK
+
+Customers need to update the `amplify.yml` to generate the DB schema if the DB is in a different VPC. Etc Etc. TK
+
+TK: npx amplify generate config --connection-string $BRANCH_CONNECTION_STRING
+
+## Add relationships between tables
+
+```ts
+import { a, defineData } from '@aws-amplify/backend'
+import { schema as rdsSchema } from './schema.rds.ts'
```

Comment:
here and in other snippets, specifying `.ts` gives the following error in my VS Code IDE with default settings:
![Screenshot 2024-04-17 at 12 22 13 PM](https://github.com/aws-amplify/docs/assets/55896475/881260b7-4b18-4ba5-9776-b97a4e5d9a7f)
We could skip that extension and that resolves it.